spark-jobs/src/main/scala/filodb/repair/PartitionKeysCopier.scala

```diff
@@ -0,0 +1,182 @@
+    }
+
+    val sourceConfigPaths: util.List[lang.String] = sourceConfig.getStringList("dataset-configs")
+    val datasetConfig: Config = sourceConfigPaths.stream()
+      .map[Config](new util.function.Function[lang.String, Config]() {
+        override def apply(path: lang.String): Config = getConfig(path)
+      })
+      .filter(new util.function.Predicate[Config] {
+        override def test(conf: Config): Boolean = conf.getString("dataset").equals(sourceDataset)
+      })
+      .findFirst()
+      .orElseThrow()
+
+    val numShards = datasetConfig.getInt("num-shards")
+    numShards
+  }
+
+  // Examples: 2019-10-20T12:34:56Z  or  2019-10-20T12:34:56-08:00
+  private def parseDateTime(str: String) = Instant.from(DateTimeFormatter.ISO_OFFSET_DATE_TIME.parse(str))
+
+  // Both "source" and "target" refer to file paths which define config files that have a
+  // top-level "filodb" section and a "cassandra" subsection.
+  private val sourceConfig = openConfig("spark.filodb.partitionkeys.copier.source.configFile")
+  private val targetConfig = openConfig("spark.filodb.partitionkeys.copier.target.configFile")
+  private val sourceCassConfig = sourceConfig.getConfig("cassandra")
+  private val targetCassConfig = targetConfig.getConfig("cassandra")
+  private val sourceDataset = conf.get("spark.filodb.partitionkeys.copier.source.dataset")
+  private val sourceDatasetRef = DatasetRef.fromDotString(sourceDataset)
+  private val targetDatasetRef = DatasetRef.fromDotString(conf.get("spark.filodb.partitionkeys.copier.target.dataset"))
+  private val sourceSession = FiloSessionProvider.openSession(sourceCassConfig)
+  private val targetSession = FiloSessionProvider.openSession(targetCassConfig)
+
+  private val numOfShards: Int = getShardNum
+  private val ingestionTimeStart = parseDateTime(conf.get("spark.filodb.partitionkeys.copier.ingestionTimeStart"))
+  private val ingestionTimeEnd = parseDateTime(conf.get("spark.filodb.partitionkeys.copier.ingestionTimeEnd"))
+  private val diskTimeToLiveSeconds = conf.getTimeAsSeconds("spark.filodb.partitionkeys.copier.diskTimeToLive")
+  private val readSched = Scheduler.io("cass-read-sched")
+  private val writeSched = Scheduler.io("cass-write-sched")
+
+  val sourceCassandraColStore = new CassandraColumnStore(sourceConfig, readSched, sourceSession)(writeSched)
+  val targetCassandraColStore = new CassandraColumnStore(targetConfig, readSched, targetSession)(writeSched)
+
+  // Destructively deletes everything in the target before updating anthing. Is used when chunks aren't aligned.
+  private[repair] val deleteFirst = conf.getBoolean("spark.filodb.partitionkeys.copier.deleteFirst", false)
+  // Disable the copy phase either for fully deleting with no replacement, or for no-op testing.
+  private[repair] val noCopy = conf.getBoolean("spark.filodb.partitionkeys.copier.noCopy", false)
+  private[repair] val splitsPerNode = conf.getInt("spark.filodb.partitionkeys.copier.splitsPerNode", 1)
```

Comment:
Use the `splitsPerNode` within the cassandra config. Remove additional spark parameter.

Reply:
@vishramachandran is this referring to the following? 

https://github.com/filodb/FiloDB/blob/70b506e0356da8997bb22951cbdc935b6615e8bb/core/src/main/resources/filodb-defaults.conf#L363

Reply:
```
    # amount of parallelism to introduce in the token scan queries. This controls number of spark partitions
    # increase if the number of splits seen in cassandra reads is low and spark jobs are slow, or
    # if we see Cassandra read timeouts in token range scans.
    num-token-range-splits-for-scans = 10
```